app/src/main/java/com/example/next_show/fragments/ShowDetailFragment.java

```diff
@@ -149,7 +167,41 @@ public void done(ParseException e) {
         });
     }
 
-    // update rating
+    private void fetchShowPoster(ImageView iv) {
+        // create HTTP client
+        AsyncHttpClient client = new AsyncHttpClient();
+        String id = currShow.getId();
+        String getUrl = SHOW_DETAIL_URL + id + ADD_API_KEY + getContext().getString(R.string.movie_api_key);
+
+        // call URL and parse through JSON
+        client.get(getUrl, new JsonHttpResponseHandler() {
+            @Override
+            public void onSuccess(int i, Headers headers, JSON json) {
+                Log.d(TAG, "onSuccess");
+
+                // get json object
+                JSONObject jsonObj = json.jsonObject;
+
+                // get results and save to show object
+                try {
+                    String path = jsonObj.getString("backdrop_path");
+                    currShow.setImageUrl(path);
+
+                    // display backdrop poster here
+                    Glide.with(getContext()).load(currShow.getImageUrl()).into(iv);
+
+                } catch (JSONException e) {
+                    Log.e(TAG, "Hit JSON Exception");
+                }
+            }
+
+            @Override
+            public void onFailure(int i, Headers headers, String s, Throwable throwable) {
+                Log.e(TAG, "Error Code: " + i, throwable);
+            }
+        });
+    }
```

Comment:
I think this might be another good case where we pull this out of here and into another class that just handles data fetching, then we have another API setup where someone can pass in a callback:

That would keep this fragment code a bit cleaner and then all it does it make the API request with a callback instance.

Something to think about -- if you already have an image url, is there any point in fetching for it again?